crates/ruma-state-res/src/events/join_rules.rs

```diff
@@ -0,0 +1,76 @@
+//! Types to deserialize `m.room.join_rules` events.
+
+use std::ops::Deref;
+
+use ruma_common::serde::{from_raw_json_value, PartialEqAsRefStr, StringEnum};
+use serde::Deserialize;
+
+use super::Event;
+
+/// A helper type for an [`Event`] of type `m.room.join_rules`.
+///
+/// This is a type that deserializes each field lazily, when requested.
+#[derive(Debug, Clone)]
+pub struct RoomJoinRulesEvent<E: Event>(E);
+
+impl<E: Event> RoomJoinRulesEvent<E> {
+    /// Construct a new `RoomJoinRulesEvent` around the given event.
+    pub fn new(event: E) -> Self {
+        Self(event)
+    }
+
+    /// The join rule of the room.
+    pub fn join_rule(&self) -> Result<JoinRule, String> {
+        #[derive(Deserialize)]
+        struct RoomJoinRulesContentJoinRule {
+            join_rule: Option<JoinRule>,
+        }
+
+        let content: RoomJoinRulesContentJoinRule =
+            from_raw_json_value(self.content()).map_err(|err: serde_json::Error| {
+                format!("invalid `join_rule` field in `m.room.join_rules` event: {err}")
+            })?;
+        content
+            .join_rule
+            .ok_or_else(|| "missing `join_rule` field in `m.room.join_rules` event".to_owned())
```

Comment:
Like with `membership` from `RoomMemberEventContent`, this probably shouldn't be an `Option`
```suggestion
        #[derive(Deserialize)]
        struct RoomJoinRulesContentJoinRule {
            join_rule: JoinRule,
        }

        let content: RoomJoinRulesContentJoinRule =
            from_raw_json_value(self.content()).map_err(|err: serde_json::Error| {
                format!("invalid or missing `join_rule` field in `m.room.join_rules` event: {err}")
            })?;
        content
            .join_rule
```

Reply:
Applied